mlflow/sklearn/utils.py

```diff
@@ -77,6 +94,219 @@ def _get_args_for_score(score_func, fit_func, fit_args, fit_kwargs):
+    # name_args_func_metrics_dict stores pairs of (function name, (function arguments, function call))
+    metrics_dict_regressor = {
+        "mse": ((y_true, y_pred), sklearn.metrics.mean_squared_error),
+        "rmse": ((y_true, y_pred), sklearn.metrics.mean_squared_error),
+        "mae": ((y_true, y_pred), sklearn.metrics.mean_absolute_error),
+        "r2_score": ((y_true, y_pred), sklearn.metrics.r2_score),
+    }
+
+    name_score_dict = {}
+    for func_name, func_args_call in metrics_dict_regressor.items():
+        try:
+            func_args = func_args_call[0]
+            func_call = func_args_call[1]
+            sample_weight = (
+                _get_sample_weight(fit_arg_names, fit_args, fit_kwargs)
+                if _SAMPLE_WEIGHT in fit_arg_names
+                else None
+            )
+            # Always add the multioutput default value 'uniform_average'
+            func_args = (*func_args, sample_weight, "uniform_average")
+
+            # For a special case of rmse, the last boolean for parameter 'squared' is needed and should be False
+            func_args = (*func_args, False) if (func_name == "rmse") else func_args
+            func_score = func_call(*func_args)
+        except Exception as e:  # pylint: disable=broad-except
+            _log_warning(func_name, func_call, e)
+        else:
+            name_score_dict[func_name] = func_score
+
+    return name_score_dict
+
+
+def _get_clusterer_metrics(trained_estimator, fit_args, fit_kwargs):
+    """
+    Compute and log various common metrics for clusterers
+
+    For (1) completeness score:
+    https://scikit-learn.org/stable/modules/generated/sklearn.metrics.completeness_score.html#sklearn.metrics.completeness_score
+    (2) homogeneity score:
+    https://scikit-learn.org/stable/modules/generated/sklearn.metrics.homogeneity_score.html#sklearn.metrics.homogeneity_score
+    (3) v-measure score:
+    https://scikit-learn.org/stable/modules/generated/sklearn.metrics.v_measure_score.html#sklearn.metrics.v_measure_score
+    By default, we choose the parameter 'beta' for v-measure score to be 1.0.
+
+    Steps:
+    1. Extract X and y_true from fit_args and fit_kwargs.
+    2. return a dictionary of metric(name, value)
+
+    :param trained_estimator: The already fitted clusterer
+    :param fit_args: Positional arguments given to fit_func.
+    :param fit_kwargs: Keyword arguments given to fit_func.
+    :return: dictionary of (function name, computed value)
+    """
+    import sklearn
+
+    print("In Clusterer!")
+    fit_arg_names = _get_arg_names(trained_estimator.fit)
+    X, y_true, y_pred = _get_fit_Xy(trained_estimator, fit_args, fit_kwargs, fit_arg_names)
+
+    # Maintain a metrics dictionary to store metrics info
+    # name_args_func_metrics_dict stores pairs of (function name, (function arguments, function call))
+    metrics_dict_clusterer = {
+        "completeness_score": ((y_true, y_pred), sklearn.metrics.completeness_score),
```

Comment:
For many clustering models, `y_true` (which corresponds to the `y` argument of `fit()`) is an unused placeholder value. For example, see https://scikit-learn.org/stable/modules/generated/sklearn.cluster.KMeans.html#sklearn.cluster.KMeans.fit_predict.

Additionally, the number of classes reflected by `y_true` does not necessarily correspond to the number of clusters. I.e. there may be fewer clusters than labels or more clusters than labels.

Accordingly, I'm not sure that `y_true` is the correct value to pass in here. Is there a different value that should be used instead?